packages/@aws-cdk/toolkit-lib/test/actions/diff.test.ts

```diff
@@ -0,0 +1,253 @@
+            newValue: {
+              Type: 'AWS::S3::Bucket',
+              UpdateReplacePolicy: 'Retain',
+              DeletionPolicy: 'Retain',
+              Metadata: { 'aws:cdk:path': 'Stack1/MyBucket/Resource' },
+            },
+          }),
+        }),
+      },
+    }));
+  });
+
+  describe('templatePath', () => {
+    test('fails with multiple stacks', async () => {
+      // WHEN + THEN
+      const cx = await builderFixture(toolkit, 'two-empty-stacks');
+      await expect(async () => await toolkit.diff(cx, { 
+        stacks: { strategy: StackSelectionStrategy.ALL_STACKS },
+        templatePath: path.join(__dirname, '..', '_fixtures', 'stack-with-bucket', 'cdk.out', 'Stack1.template.json'),
+      })).rejects.toThrow(/Can only select one stack when comparing to fixed template./);
+    });
+
+    test('fails with bad file path', async () => {
+      // WHEN + THEN
+      const cx = await builderFixture(toolkit, 'stack-with-bucket');
+      await expect(async () => await toolkit.diff(cx, { 
+        stacks: { strategy: StackSelectionStrategy.ALL_STACKS },
+        templatePath: path.join(__dirname, 'blah.json'),
+      })).rejects.toThrow(/There is no file at/);
+    });
+
+    test('returns regular diff', async () => {
+      // WHEN
+      const cx = await builderFixture(toolkit, 'stack-with-bucket');
+      const result = await toolkit.diff(cx, {
+        stacks: { strategy: StackSelectionStrategy.ALL_STACKS },
+        templatePath: path.join(__dirname, '..', '_fixtures', 'two-empty-stacks', 'cdk.out', 'Stack1.template.json'),
+      });
+
+      // THEN
+      expect(result).toMatchObject(expect.objectContaining({
+        resources: {
+          diffs: expect.objectContaining({
+            MyBucketF68F3FF0: expect.objectContaining({
+              isAddition: true,
+              isRemoval: false,
+              oldValue: undefined,
+              newValue: {
+                Type: 'AWS::S3::Bucket',
+                UpdateReplacePolicy: 'Retain',
+                DeletionPolicy: 'Retain',
+                Metadata: { 'aws:cdk:path': 'Stack1/MyBucket/Resource' },
+              },
+            }),
+          }),
+        },
+      }));
+    });
+
+    test('returns security diff', async () => {
+      // WHEN
+      const cx = await builderFixture(toolkit, 'stack-with-role');
+      const result = await toolkit.diff(cx, {
+        stacks: { strategy: StackSelectionStrategy.ALL_STACKS },
+        securityOnly: true,
+        templatePath: path.join(__dirname, '..', '_fixtures', 'two-empty-stacks', 'cdk.out', 'Stack1.template.json'),
+      });
+
+      // THEN
+      expect(ioHost.notifySpy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'diff',
+        level: 'warn',
+        code: 'CDK_TOOLKIT_W0000',
+        message: expect.stringContaining('This deployment will make potentially sensitive changes according to your current security approval level (--require-approval broadening)'),
+      }));
```

Comment:
sanity check: do we expect to have this as a warning notification when we run `toolkit.diff`?

Reply:
No, probably not.

Reply:
this comes from [here](https://github.com/aws/aws-cdk-cli/blob/af827ea95df07f6090d69aa9119043f642f72c2a/packages/%40aws-cdk/tmp-toolkit-helpers/src/api/diff/diff.ts#L79-L80)

it was previously a `warning()` before being refactored into `ioDefaultHelper.warning`, so it was always printed as a warning even when the command was `cdk diff` (and not `cdk deploy`, where presumably you would need to stop and confirm the diff). `cdk deploy` has its own confirm message that actually prompts the user.

i think we should be able to just delete those two lines -- i don't think they are useful. alternatively, i don't think its harmful to leave them in.